### DISCUSSION TITLE
printed file using Microsoft's pdf printer is too large in size

### DISCUSSION THREAD

**Discussion by medongaw** [sumatrapdfreader/sumatrapdf#2497] [Q&A]
I think this is caused by the sumatra pdf not being able to recognize the text as text so it just handles the printing process as if it is a scanned document, therefore the file size drastically increases by 10-30 times.

I guess this is not a problem of microsoft printer because other applications I tried printed it as pdf with no size enlargement issue.

is this normally expected behavior? thanks

**Comment by GitHubRulesOK** [comment]
yes printing is done by windows print rendering as image regardless if it is an image or not, it is the only consistent way for a multi-format app to work in the majority of cases. the option to attempt otherwise was removed after version 3.0 had many print related issues due to the increasing types of driver methods. Each print driver has its own way of handling PAI so for example MS Print to PDF has a default of fuzzy 600dpi which generates enormous outputs. You could try reduce to more normal 300dpi and the output will be 25% of default

**Comment by wliceberg** [comment]
hi,Has your problem been solved? 
I use the command line 'sumatrapdf.exe -print-to printer-name -print-settings settings-list filepath' to send a PDF file to the printer. 
The PDF file size is 7M. To printer size is 2G.

Is there a parameter that can change the size？

**Reply by GitHubRulesOK** [reply]
the size is determined by the printers driver this is an extreme case
Input as PDF
```
File: C.pdf
File Size: 7.92 MB (8,308,342 Bytes)
Number of Pages: 80
Page Size: 17.71 x 25.86 cm
true page size is 2093 x 3056  Pixels (6.40 MPixels)
``` 
![image](https://user-images.githubusercontent.com/12754913/168604091-214bf824-6a5b-4067-9576-dc14fac5d4dc.png)


TIFF Output as Microsoft Printed FaX with larger sheets As G3 mono we can expect castellation thus inferior output.
```
File: 1D8691F82ADCA.tif
File Size: 3.26 MB (3,419,594 Bytes)
Number of Pages: 80
Page Size: 21.52 x 29.12 cm
***Fax is 200-203 dpi hence the bad edges***
```
![image](https://user-images.githubusercontent.com/12754913/168603775-28b6c68f-87e4-4f46-b9dc-ec39badb2c25.png)


BMP Using My Microsoft PDF
```
File: C:\Users\K\Documents\printout.pdf
PDF Producer: Microsoft: Print To PDF
File Size: 128.47 MB (134,707,074 Bytes)
Number of Pages: 80
Page Size: 21.0 x 29.7 cm (A4)
Pages are roughly quadrupled in size 4183 x 6108  Pixels (25.55 MPixels) 
but just one page is now 100 times bigger 73.10 MB (76,649,309 Bytes)
```
![image](https://user-images.githubusercontent.com/12754913/168604529-4d0e3f89-1a34-41e0-9b14-57104ecb9382.png)


PNG Using MS XPS printer to convert into 80 x PNG at A4 the pixel scaling gets larger but quality is reasonable as lossless
```
File: test.oxps
File Size: 164.09 MB (172,055,739 Bytes)
Number of Pages: 80
Page Size: 20.98 x 29.69 cm (A4)
as separate images we see why the file is so big each page is now
Page Size: 131.26 x 185.63 cm

```
![image](https://user-images.githubusercontent.com/12754913/168605042-47d3bed5-9542-489f-b747-472387a351c6.png)


JPEG using MS XPS at max compressed quality, actually seems there is some mis-labeling?? about the content
```
File: test2.oxps
File Size: 87.68 MB (91,937,851 Bytes)
Number of Pages: 80
Page Size: 20.98 x 29.69 cm (A4)

individual pages are PNG not jpeg !!

File: jpeg.oxps\Documents\1\Resources\Images\1.PNG
File Size: 610.03 KB (624,671 Bytes)
Number of Pages: 1
Page Size: 43.59 x 63.64 in (110.7186 x 161.6456 cm)
thus only the pixel scale is reduced from 131.26 x 185.63 cm not the actual quality !!

````
![image](https://user-images.githubusercontent.com/12754913/168603881-2a09514a-637b-4dde-94b8-a7e5a0b35510.png)
